purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -46,15 +49,24 @@ internal class PostReceiptHelper(
             receiptInfo,
             storeUserID,
             marketplace,
-            onSuccess = { _, _ ->
+            onSuccess = {
                 deviceCache.addSuccessfullyPostedToken(purchaseToken)
                 onSuccess()
             },
-            onError = { error, shouldConsumePurchase, _, _ ->
+            onError = { error, shouldConsumePurchase, isServerError, _ ->
                 if (shouldConsumePurchase) {
```

Comment:
I think it's confusing that you're not checking if isServerError here because `shouldConsumePurchase` will already be sent as false here if offlineEntitlements shouldn't be used

Reply:
Yeah, this is pretty confusing and error prone... Will try to refactor this so it's clearer that those 2 flags are exclusive

Reply:
I tried an approach to making this clearer in this commit: https://github.com/RevenueCat/purchases-android/pull/964/commits/d2f64cf397977229047eedf5111cfd415b070652. Lmk what you think!